test/e2e/storage/testsuites/provisioning.go

```diff
@@ -45,6 +45,7 @@ import (
 // Not all parameters are used by all tests.
```

Comment:
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/testsuites/provisioning.go#L388,
```framework.ExpectNoError(e2epv.WaitForPersistentVolumeDeleted(client, pv.Name, 5*time.Second, 20*time.Minute))```
there is a wait for pv delete step that has 20 mins timeout not sure why. This needs to be revisited, perhaps in a different PR.

Reply:
I didn't add the comment, I just moved it to a new place :-)

I had to trace back quite a bit to where this comment was originally added: https://github.com/kubernetes/kubernetes/commit/46fb4172999bebcf822a2953e1783ed0428adc70

I think the 20 mins is to also account for the time period that it may take to unmount and detach the volume. 20 minutes seems quite long and could probably be part of the customizable timeouts?